experimental/DoubleComplex/src/Attributes.jl

```diff
@@ -129,23 +128,23 @@ function vertical_range(D::DoubleComplexOfMorphisms)
 end
 
 function right_bound(D::DoubleComplexOfMorphisms) 
-  isdefined(D, :right_bound) || error("complex has no known horizontal upper bound")
-  return D.right_bound
+  D.right_bound !== nothing || error("complex has no known horizontal upper bound")
+  return D.right_bound::Int
```

Comment:
This annotation shouldn't be necessary, as the compiler can prove using the previous line that `Nothing` is ruled out. That said, it also doesn't hurt, so it's fine to leave it in.